Sources/App/configure.swift

```diff
@@ -38,7 +38,7 @@ public func configure(_ config: inout Config, _ env: inout Environment, _ servic
 
     services.register(OnTapController.self)
     services.register(KarmaController.self)
-    services.register(try Secrets.detect())
+    services.register(try Secrets.detect(), as: Secrets.self)
```

Comment:
This is better, not perfect I guess but you are right that it doesn't really feel like there is a "right" way to do it. This is more explicit which I like